### PR TITLE
[ci] rspec: Replace mocha with rspec-mocks

### PR DESCRIPTION
**Pull request by bgeuken**
Once the old test suite is gone, we can even remove the mocha gem
dependency.

https://relishapp.com/rspec/rspec-mocks/v/3-5/docs/basics
https://github.com/rspec/rspec-mocks

### CODE REVIEW

File: src/api/spec/controllers/webui/package_controller_spec.rb [hunk 1]
```diff
@@ -229,7 +229,7 @@
 
     context 'without build results' do
       before do
-        Buildresult.stubs(:find_hashed).returns(nil)
+        allow(Buildresult).to receive(:find_hashed).and_return(nil)
```
Comment (Ana06) [inline src/api/spec/controllers/webui/package_controller_spec.rb]: the `.and_return(nil)` is not needed, just:

`allow(Buildresult).to receive(:find_hashed)`

File: src/api/spec/controllers/webui/patchinfo_controller_spec.rb [hunk 2]
```diff
@@ -78,7 +78,7 @@ def do_proper_post_save
 
     context 'when the patchinfo package file is not found' do
       before do
-        Package.any_instance.stubs(:patchinfo).returns(nil)
+        allow_any_instance_of(Package).to receive(:patchinfo).and_return(nil)
```
Comment (Ana06) [inline src/api/spec/controllers/webui/patchinfo_controller_spec.rb]: the `.and_return(nil)` is not needed, just:

`allow_any_instance_of(Package).to receive(:patchinfo)`

File: src/api/spec/controllers/webui/package_controller_spec.rb (hunk 1 above)
Comment (bgeuken) [inline src/api/spec/controllers/webui/package_controller_spec.rb]: Good catch

**Comment by hennevogel** [comment]
Cool. LGTM!

**Pull request merged**